quickwit/quickwit-metastore/src/metastore/file_backed_metastore/mod.rs

```diff
@@ -145,21 +145,21 @@ impl FileBackedMetastore {
         &self,
         index_id: &str,
         mutate_fn: impl FnOnce(&mut FileBackedIndex) -> crate::MetastoreResult<bool>,
-    ) -> MetastoreResult<()> {
+    ) -> MetastoreResult<bool> {
         let mut locked_index = self.get_locked_index(index_id).await?;
         let mut index = locked_index.clone();
         let mutation_occurred = mutate_fn(&mut index)?;
         if !mutation_occurred {
-            return Ok(());
+            return Ok(false);
         }
 
         let put_result = put_index(&*self.storage, &index).await;
         match put_result {
             Ok(()) => {
                 *locked_index = index;
-                Ok(())
+                Ok(true)
             }
-            err @ Err(_) => {
```

Comment:
Arguable. While less used and somewhat less readable,  `err @ Err(_)` clearly indicates that the wrapped error won't be used.

Reply:
Sorry, I am not really understanding the feedback here.
The error is captured just to be returned, so I felt destructing and wrapping again makes it easier to read.

Reply:
Ok got you now. In this particular case, you cannot however just forward the bound `err` thought because its type is not the Error, it is the whole Result we are binding and this result type does not match the function return type.
in fact this `err @ Err(_)`  should be `result @ Err(_)`